File: src/offer15.numberof1/numberof1.cpp
class Solution1 {
public:
     int  NumberOf1(int n) {
         int count = 0;
         unsigned int flag = 1;
         while(flag){
             if(n & flag){
                 count++;
             }
             flag = flag << 1;
         }
         return count;
     }
};
class Solution2 {
public:
     int  NumberOf1(int n) {
         int count = 0;
         while(n){
             int nn = n - 1;
             n = n & nn;
             count++;
         }
         
         return count;
     }
};

File: src/offer15.numberof1/numberof1.java
public class Solution {
    public int NumberOf1(int n) {
        int bit = 1;
        int count = 0;
        for(int i = 0; i < 32; i++){
            if((bit & n) != 0){
                count++;
            }
            bit <<= 1;
        }
        return count;
    }
}

public class Solution {
    public int NumberOf1(int n) {
        int count = 0;
        while(n != 0){       //!! 不是大于0
            int n1 = n - 1;
            n = n1 & n;
            count++;
        }
        return count;
    }
}